### ISSUE TITLE
memory usage problem

### ISSUE THREAD

**Issue by leepeter2019** [vusec/TIFF#4]
I activate the taint_detect module which was turn offed.
But when I run pin tool with taint_detect module, it run out of memory. 
The taint_detect module uses too much memory allocation and there might not release memory usage. 
Have you ever same experience when you run the taint_detect module?
I cannot find which function have the memory usage problem.
Any suggestions?

**Comment by vivek425ster** [comment]
@leepeter2019 Yes taint detect has known issues with memory. I would suggest using a higher RAM system for running the system.
